tests/IBTK/var_c_helmholtz_2d.cpp

```diff
@@ -0,0 +1,261 @@
+        SAMRAIVectorReal<NDIM, double> f_vec("f", patch_hierarchy, 0, patch_hierarchy->getFinestLevelNumber());
+        SAMRAIVectorReal<NDIM, double> e_vec("e", patch_hierarchy, 0, patch_hierarchy->getFinestLevelNumber());
+        SAMRAIVectorReal<NDIM, double> r_vec("r", patch_hierarchy, 0, patch_hierarchy->getFinestLevelNumber());
+
+        u_vec.addComponent(u_cc_var, u_cc_idx, h_cc_idx);
+        f_vec.addComponent(f_cc_var, f_cc_idx, h_cc_idx);
+        e_vec.addComponent(e_cc_var, e_cc_idx, h_cc_idx);
+        r_vec.addComponent(r_cc_var, r_cc_idx, h_cc_idx);
+
+        u_vec.setToScalar(0.0);
+        f_vec.setToScalar(0.0);
+        e_vec.setToScalar(0.0);
+        r_vec.setToScalar(1.0);
+
+        // Setup exact solutions.
+        muParserCartGridFunction u_fcn("u", app_initializer->getComponentDatabase("u"), grid_geometry);
+        muParserCartGridFunction f_fcn("f", app_initializer->getComponentDatabase("f"), grid_geometry);
+
+        u_fcn.setDataOnPatchHierarchy(e_cc_idx, e_cc_var, patch_hierarchy, 0.0);
+        f_fcn.setDataOnPatchHierarchy(f_cc_idx, f_cc_var, patch_hierarchy, 0.0);
+
+        // Ensure that the right-hand-side vector has no components in the
+        // nullspace of the operator.
+        f_vec.addScalar(Pointer<SAMRAIVectorReal<NDIM, double> >(&f_vec, false),
+                        -f_vec.dot(Pointer<SAMRAIVectorReal<NDIM, double> >(&r_vec, false)) /
+                            r_vec.dot(Pointer<SAMRAIVectorReal<NDIM, double> >(&r_vec, false)));
+
+        // Defining variable C coefficient patch data index
+        for (int ln = 0; ln <= patch_hierarchy->getFinestLevelNumber(); ln++)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+
+            for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+            {
+                Pointer<Patch<NDIM> > patch = level->getPatch(p());
+                const Box<NDIM>& patch_box = patch->getBox();
+                Pointer<CellData<NDIM, double> > C_data = patch->getPatchData(C_coef_idx);
+                for (Box<NDIM>::Iterator it(patch_box); it; it++)
+                {
+                    CellIndex<NDIM> ci(it());
+                    IBTK::VectorNd cell_coord;
+                    Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+                    const double* patch_X_lower = patch_geom->getXLower();
+                    const SAMRAI::hier::Index<NDIM>& patch_lower_index = patch_box.lower();
+                    const double* const patch_dx = patch_geom->getDx();
+                    for (unsigned int d = 0; d < NDIM; ++d)
+                    {
+                        cell_coord[d] = patch_X_lower[d] + patch_dx[d] * (ci(d) - patch_lower_index(d) + 0.5);
+                    }
+                    (*C_data)(ci) = std::pow(cell_coord[0], 2.0) + std::pow(cell_coord[1], 2.0);
+                }
+            }
+        }
```

Comment:
@Raamkrishnan : Can you use muParserCartGridFunction here instead of coding `C coef`?